src/levanter/distributed.py

```diff
@@ -276,7 +276,12 @@ def _munge_address_port(address: str):
             else:
                 logger.warning(f"Failed to initialize ray with address {address}. Retrying...")
                 continue
-    atexit.register(lambda: ray.shutdown())
+
+    def do_shutdown():
```

Comment:
i should maybe make it so that the auto-ray-cluster stuff maybe doesn't run if we're already in a ray cluster, and instead connects to the outer ray cluster...